tests/telemetry/events/run_events_test.py

```diff
@@ -26,3 +26,4 @@ def run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, trigger, json
     dest = "/tmp/" + json_file
     duthost.copy(src=op_file, dest=dest)
     validate_yang(duthost, dest, tag)
+    time.sleep(1)  # provide buffer for multiple calls to py_gnmicli
```

Comment:
>sleep [](http://example.com/codeflow?start=9&length=5)

If one platform is slow, 1s will be not enough? Always try to prevent fix time sleep.

Reply:
From my testing, 1 second was enough of a buffer between tests such that gnmi_cli process can exit from previous testcase and new testcase will run cleanly with trigger, then starting up new gnmi_cli process to listen.

Reply:
If you intention is to wait "gnmi_cli process can exit from previous testcase", implement a wait for that event. This is the most robust way to wait.

I do not know what will happen if testing on a weak hardware. Or, if you change time interval to <1s, is any possibly to fail the test? If yes, we have a timing risk.

Reply:
Added wait_until and removed time.sleep